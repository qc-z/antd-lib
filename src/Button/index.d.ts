import { MyButtonProps } from "./button";

declare const AyButton: React.ForwardRefExoticComponent<
  MyButtonProps & React.RefAttributes<HTMLDivElement>
>;

export default AyButton;
