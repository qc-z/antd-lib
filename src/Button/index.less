.my-button {
  &.simple {
    width: 100%;
  }

  &.sub {
    height: unset;
    padding: 0;
    color: #00000073;
    line-height: 1em;
    background-color: unset;
    border: unset;
    box-shadow: unset;

    &:hover,
    &:focus {
      color: #000000d9;
      background-color: unset;
    }
  }
}
