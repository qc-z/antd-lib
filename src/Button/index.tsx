import React from "react";
import { Button } from "antd";
// import "antd/es/button/style";

const App = () => (
  <div className="App">
    <Button type="primary">Button</Button>
  </div>
);

export default App;
