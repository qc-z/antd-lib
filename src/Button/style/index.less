@import 'antd/es/Button/style/index.less';
@import '../../style/custom.less';
@import './button.less';
