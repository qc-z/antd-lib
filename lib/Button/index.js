"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("antd/es/button/style");

var _button = _interopRequireDefault(require("antd/es/button"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import "antd/es/button/style";
var App = function App() {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "App"
  }, /*#__PURE__*/_react.default.createElement(_button.default, {
    type: "primary"
  }, "Button"));
};

var _default = App;
exports.default = _default;