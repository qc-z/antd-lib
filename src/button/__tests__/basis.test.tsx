import renderer from 'react-test-renderer'
import Button from '../button'
test('按钮', () => {
  const component = renderer.create(
    <Button type="primary">按钮</Button>
  )
  const button = toJson(component)
  expect(button).toMatchSnapshot()

  // manually trigger the callback
  // tree.props.onMouseEnter()

  // re-rendering
  // tree = toJson(component)
  // expect(tree).toMatchSnapshot()

  // manually trigger the callback
})
function toJson(
  component: renderer.ReactTestRenderer
) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}
