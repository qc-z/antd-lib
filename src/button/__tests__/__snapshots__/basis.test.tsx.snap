// Vitest Snapshot v1

exports[`按钮 1`] = `
<button
  className="ant-btn ant-btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span>
    按 钮
  </span>
</button>
`;
